encodings/alp/src/alp/mod.rs

```diff
@@ -157,6 +157,13 @@ pub trait ALPFloat: private::Sealed + Float + Display + 'static {
         values
     }
 
+    fn decode_vec(encoded: Vec<Self::ALPInt>, exponents: Exponents) -> Vec<Self> {
+        encoded
+            .into_iter()
+            .map(move |encoded| Self::decode_single(encoded, exponents))
```

Comment:
What I think it comes down to is this `move`. It is much slower without it, I'm wondering if it's because the closure doesn't get inlined when it only borrows its scope?

Reply:
Or perhaps the move removes a pointer dereference of a borrowed exponents struct which instead is copied?